### PR TITLE
[BugFix]fix array_map crash

### LINKED ISSUE #49730: [Bug] BE crash when using ARRAY_MAP and ARRAY_GENERATE
### Steps to reproduce the behavior (Required)
1. CREATE TABLE
```
CREATE TABLE B
(
id INT NOT NULL COMMENT "",
data_array ARRAY<STRING> NULL COMMENT ""
) ENGINE=OLAP
PRIMARY KEY(`id`)
DISTRIBUTED BY HASH(`id`) BUCKETS 1
PROPERTIES (
"replication_num" = "1"
);
```
2. INSERT INTO 
```
INSERT INTO B (id, data_array)
VALUES 
(1, ARRAY_MAP(
    x -> CAST(x AS STRING), 
    ARRAY_GENERATE(1, 1000)
)),
(2, ARRAY_MAP(
    x -> CAST(x AS STRING), 
    ARRAY_GENERATE(1, 1000)
)),
(3, ARRAY_MAP(
    x -> CAST(x AS STRING), 
    ARRAY_GENERATE(1, 1000)
)),
(4, ARRAY_MAP(
    x -> CAST(x AS STRING), 
    ARRAY_GENERATE(1, 1000)
)),
(5, ARRAY_MAP(
    x -> CAST(x AS STRING), 
    ARRAY_GENERATE(1, 1000)
));
```

### Expected behavior (Required)
Success

### Real behavior (Required)
Be crash:
```
*** Aborted at 1723503069 (unix time) try "date -d @1723503069" if you are using GNU date ***
PC: @          0x51c5d75 starrocks::ArrayMapExpr::evaluate_checked()
*** SIGSEGV (@0x0) received by PID 2861 (TID 0x7f855594b700) from PID 0; stack trace: ***
    @          0x6bb1782 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f85d378f630 (unknown)
    @          0x51c5d75 starrocks::ArrayMapExpr::evaluate_checked()
    @          0x4890e03 starrocks::ExprContext::evaluate()
    @          0x489114f starrocks::ExprContext::evaluate()
    @          0x3a9254c starrocks::pipeline::UnionConstSourceOperator::pull_chunk()
    @          0x3a43d1d starrocks::pipeline::PipelineDriver::process()
    @          0x3a35c7e starrocks::pipeline::GlobalDriverExecutor::_worker_thread()
    @          0x305182c starrocks::ThreadPool::dispatch_thread()
    @          0x304b4aa starrocks::Thread::supervise_thread()
    @     0x7f85d3787ea5 start_thread
    @     0x7f85d2b88b0d __clone
    @                0x0 (unknown)
```

### StarRocks version (Required)
 - branch-3.2-d35c9a7

### PR DESCRIPTION
**Pull request by before-Sunrise**
## Why I'm doing:
many exprs can not handle situation that chunk is nullptr.
And ExprContext::evaluate and VectorizedLiteral::evaluate_checked can not handle situation that chunk is empty if this expr is const

## What I'm doing:
we should at least give a dummy chunk with one row one column if this expr is constant(Previously we would provide nullptr)

Fixes [#issue](https://github.com/StarRocks/starrocks/issues/49730)

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

## Documentation PRs only:

If you are submitting a PR that adds or changes English documentation and have not
included Chinese documentation, then you can check the box to request GPT to translate the
English doc to Chinese. Please ensure to uncheck the **Do not translate** box if translation is needed.
The workflow will generate a new PR with the Chinese translation after this PR is merged.

- [ ] Yes, translate English markdown files with GPT
- [x] Do not translate

### CODE REVIEW

File: be/src/exec/pipeline/set/union_const_source_operator.cpp [hunk 1]
```diff
@@ -36,8 +38,9 @@ StatusOr<ChunkPtr> UnionConstSourceOperator::pull_chunk(starrocks::RuntimeState*
             // Each const_expr_list is projected to ONE dest row.
             DCHECK_EQ(_const_expr_lists[_next_processed_row_index + row_i].size(), columns_count);
 
+            ChunkPtr dummyChunk = ChunkHelper::createDummyChunk();
             ASSIGN_OR_RETURN(ColumnPtr src_column,
-                             _const_expr_lists[_next_processed_row_index + row_i][col_i]->evaluate(nullptr));
+                             _const_expr_lists[_next_processed_row_index + row_i][col_i]->evaluate(dummyChunk.get()));
```
Comment (satanson) [inline be/src/exec/pipeline/set/union_const_source_operator.cpp:43]: it seems that a empty_chunk is just OK;
```
ChunkPtr empty_chunk = std::make_shared<Chunk>();
```

Comment (before-Sunrise) [reply be/src/exec/pipeline/set/union_const_source_operator.cpp:43]: StatusOr<ColumnPtr> ExprContext::evaluate(Expr* e, Chunk* chunk, uint8_t* filter) {
    DCHECK(_prepared);
    DCHECK(_opened);
    DCHECK(!_closed);
#ifndef NDEBUG
    if (chunk != nullptr) {
        chunk->check_or_die();
        CHECK(!chunk->is_empty());
    }
    There is a check, do you have any context why?

Comment (before-Sunrise) [reply be/src/exec/pipeline/set/union_const_source_operator.cpp:43]: > it seems that a empty_chunk is just OK;
> 
> ```
> ChunkPtr empty_chunk = std::make_shared<Chunk>();
> ```

StatusOr ExprContext::evaluate(Expr* e, Chunk* chunk, uint8_t* filter) {
DCHECK(_prepared);
DCHECK(_opened);
DCHECK(!_closed);
#ifndef NDEBUG
if (chunk != nullptr) {
chunk->check_or_die();
CHECK(!chunk->is_empty());
}
There is a check, do you have any context why?

**Comment by ZiheLiu** [comment]
`Expr::evaluate_const` also calls `evaluate(this, nullptr)`, will it have the same problem?

```cpp
StatusOr<ColumnPtr> Expr::evaluate_const(ExprContext* context) {
    if (!is_constant()) {
        return nullptr;
    }

    if (_constant_column.ok() && _constant_column.value()) {
        return _constant_column;
    }

    // prevent _constant_column from being assigned by multiple threads in pipeline engine.
    std::call_once(_constant_column_evaluate_once,
                   [this, context] { this->_constant_column = context->evaluate(this, nullptr); });
    return _constant_column;
}
```

**Comment by before-Sunrise** [comment]
> `Expr::evaluate_const` also calls `evaluate(this, nullptr)`, will it have the same problem?
> 
> ```c++
> StatusOr<ColumnPtr> Expr::evaluate_const(ExprContext* context) {
>     if (!is_constant()) {
>         return nullptr;
>     }
> 
>     if (_constant_column.ok() && _constant_column.value()) {
>         return _constant_column;
>     }
> 
>     // prevent _constant_column from being assigned by multiple threads in pipeline engine.
>     std::call_once(_constant_column_evaluate_once,
>                    [this, context] { this->_constant_column = context->evaluate(this, nullptr); });
>     return _constant_column;
> }
> ```

I think we should handle it in ExprContext, so no place will be forgot

**Review by satanson — DISMISSED** [review]

**Review by silverbullet233 — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

**Review by satanson — APPROVED** [review]

File: be/src/exprs/expr_context.cpp [hunk 2]
```diff
@@ -177,7 +186,7 @@ StatusOr<ColumnPtr> ExprContext::evaluate(Expr* e, Chunk* chunk, uint8_t* filter
             ASSIGN_OR_RETURN(ptr, e->evaluate_with_filter(this, chunk, filter));
         }
         DCHECK(ptr != nullptr);
-        if (chunk != nullptr && 0 != chunk->num_columns() && ptr->is_constant()) {
+        if (chunk != nullptr && 0 != chunk->num_columns() && ptr->is_constant() && (dummy_chunk.get() == nullptr)) {
```
Comment (wyb) [inline be/src/exprs/expr_context.cpp:189]:
```suggestion
        if (chunk != nullptr && 0 != chunk->num_columns() && ptr->is_constant() && dummy_chunk.get() == nullptr) {
```

**Review by wyb — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**